pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/manager/UnloadManager.java

```diff
@@ -92,17 +153,35 @@ public void handleEvent(String serviceUnit, ServiceUnitStateData data, Throwable
             if (log.isDebugEnabled()) {
                 log.debug("Handling {} for service unit {} with exception.", data, serviceUnit, t);
             }
-            this.complete(serviceUnit, t);
+            complete(serviceUnit, t);
             return;
         }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling {} for service unit {}", data, serviceUnit);
+        }
         ServiceUnitState state = ServiceUnitStateData.state(data);
```

Comment:
Actually, we call this `handleEvent` after completing the future of `closeServiceUnit` now.  However, we want to call `recordReleaseLatency` before `closeServiceUnit` to include `closeServiceUnit` time in the latency.

This means we probably need to add `StateChangeListeners::notifyOnArrival` and call that at the beginning like the one below
``` java
private void handleReleaseEvent(String serviceUnit, ServiceUnitStateData data) {

        stateChangeListeners.notifyOnArrival(..) //notify first to start measure latency
        if (isTargetBroker(data.sourceBroker())) {
            ServiceUnitStateData next;
            if (isTransferCommand(data)) {
                next = new ServiceUnitStateData(
                        Assigning, data.dstBroker(), data.sourceBroker(), getNextVersionId(data));
                // TODO: when close, pass message to clients to connect to the new broker
            } else {
                next = new ServiceUnitStateData(
                        Free, null, data.sourceBroker(), getNextVersionId(data));
            }
            stateChangeListeners.notifyOnCompletion(closeServiceUnit(serviceUnit)
                            .thenCompose(__ -> pubAsync(serviceUnit, next)), serviceUnit, data)
                    .whenComplete((__, e) -> log(e, serviceUnit, data, next));
        }
    }
```